src/lib.rs

```diff
@@ -463,6 +474,219 @@ impl TextEmbedding {
+                    Array::from_shape_vec((batch_size, encoding_length), ids_array)?;
+
+                let attention_mask_array =
+                    Array::from_shape_vec((batch_size, encoding_length), mask_array)?;
+
+                let token_type_ids_array =
+                    Array::from_shape_vec((batch_size, encoding_length), typeids_array)?;
+
+                let mut session_inputs = ort::inputs![
+                    "input_ids" => Value::from_array(inputs_ids_array)?,
+                    "attention_mask" => Value::from_array(attention_mask_array)?,
+                ]?;
+                if self.need_token_type_ids {
+                    session_inputs
+                        .insert("token_type_ids", Value::from_array(token_type_ids_array)?);
+                }
+
+                let outputs = self.session.run(session_inputs)?;
+
+                let outputs = outputs["logits"].extract_tensor::<f32>().unwrap();
+
+                let scores: Vec<f32> = outputs
+                    .view()
+                    .slice(s![.., 0])
+                    .rows()
+                    .into_iter()
+                    .map(|row| row.to_vec())
+                    .flatten()
+                    .collect();
+
+                Ok(scores)
+            })
+            .flat_map(|result| result.unwrap())
+            .collect();
+
+        Ok(output)
+    }
+}
+
+/// sorted rerank result
+#[derive(Debug)]
+pub struct RerankResult {
+    sorted_texts: Vec<(String, String)>,
+    sorted_scores: Vec<f32>,
+}
+
+impl RerankResult {
+
+    /// returns a collection of text pairs where the scores are greater than the given threshold.
+    fn gt(&self, s: f32) -> Result<Vec<(&str, &str)>> {
+        let mut result = Vec::new();
+        for ((t1, t2), score) in self.sorted_texts.iter().zip(self.sorted_scores.iter()) {
+            if *score > s {
+                result.push((t1.as_str(), t2.as_str()));
+            } else {
+                break;
+            }
+        }
+        Ok(result)
+    }
+
+    /// return the texts which limit with size
+    fn limit(&self, size: usize) -> Result<Vec<(&str, &str)>> {
+        assert!(size <= self.sorted_texts.len(), "size exceeds the length of sorted_texts");
+
+        Ok(self.sorted_texts
+            .iter()
+            .take(size)
+            .map(|(s1, s2)| (s1.as_str(), s2.as_str()))
+            .collect())
+    }
+
+    /// return the texts which score larger than s and limit with size
+    fn gt_limit(&self, s: f32, size: usize) -> Result<Vec<(&str, &str)>>{
+        assert!(size <= self.sorted_texts.len(), "size exceeds the length of sorted_texts");
+
+        let mut result = Vec::new();
+        for ((t1, t2), score) in self.sorted_texts.iter().zip(self.sorted_scores.iter()) {
+            if *score > s && result.len() < size {
+                result.push((t1.as_str(), t2.as_str()));
+            } else {
+                break;
+            }
+        }
+        Ok(result)
+    }
+}
```

Comment:
I recommend not providing helpers like these. Since they're trivial and can be implemented by users as needed.

Reply:
Yep, i will reomve this in the next commit.

Reply:
Sorry if I wasn't clear.
I meant only the `RerankResult` `impl`s.